src/core/lib/surface/server.h

```diff
@@ -448,7 +434,10 @@ class Server : public InternallyRefCounted<Server>,
   bool starting_ ABSL_GUARDED_BY(mu_global_) = false;
   CondVar starting_cv_;
 
-  std::vector<std::unique_ptr<RegisteredMethod>> registered_methods_;
+  std::unordered_map<uint32_t /*hash of host and method*/,
+                     std::vector<std::unique_ptr<
```

Comment:
@ctiller do we need a vector for each hash or can we assume there is only one registered method per host/method combo?